pandas/sparse/tests/test_array.py

```diff
@@ -62,6 +66,10 @@ def test_constructor_copy(self):
         not_copy.sp_values[:3] = 0
         self.assertTrue((self.arr.sp_values[:3] == 0).all())
 
+    def test_constructor_match_dtype(self):
+        res = SparseArray(self.barr_data, dtype=bool)
+        self.assertEqual(res.dtype, bool)
```

Comment:
actually construct what it should look like as an expected and use ``assert_sp_array_equal`` (which might need the same change to use ``isnull`` above)